pkg/cloud/gcp/actuators/machine/reconciler_test.go

```diff
@@ -175,6 +177,107 @@ func TestCreate(t *testing.T) {
 				}
 			},
 		},
+		{
+			name: "a2 instance create succeeds when quota is available",
+			providerSpec: &machinev1.GCPMachineProviderSpec{
+				Region:      "test-region",
+				Zone:        "test-zone",
+				MachineType: "a2-highgpu-4g",
+				Disks: []*machinev1.GCPDisk{
+					{
+						Boot:  true,
+						Image: "projects/fooproject/global/images/uefi-image",
+					},
+				},
+			},
+			mockGPUCompatibleMachineTypesList: func(project string, zone string, ctx context.Context) (map[string]computeservice.GpuInfo, []string) {
+				var compatibleMachineType = []string{}
+				var gpuInfo = map[string]computeservice.GpuInfo{
+					"a2-highgpu-4g": computeservice.GpuInfo{
+						Type:  "nvidia-a100-80gb",
+						Count: 1,
+					},
+				}
+				return gpuInfo, compatibleMachineType
+			},
+			mockRegionGet: func(project string, region string) (*compute.Region, error) {
+				var computeQuota = &compute.Quota{
+					Limit:  1.0,
```

Comment:
It looks like the limit is the only that's changing between the implementations of `mockRegionGet`; could we simplify the test struct to only include the limit and then have a shared method?

Reply:
sure, i can change it. i was following the previous pattern for injecting mock functions into the test structure.

Reply:
Discussed at standup - the code as-is works and covers more cases.